ts/Series/Treegraph/TreegraphSeries.ts

```diff
@@ -0,0 +1,659 @@
+/* *
+ *
+ *  (c) 2010-2022 Pawel Lysy Grzegorz Blachlinski
+ *
+ *  License: www.highcharts.com/license
+ *
+ *  !!!!!!! SOURCE GETS TRANSPILED BY TYPESCRIPT. EDIT TS FILE ONLY. !!!!!!!
+ *
+ * */
+
+'use strict';
+
+/* *
+ *
+ *  Imports
+ *
+ * */
+
+import type TreegraphPoint from './TreegraphPoint';
+import type TreegraphSeriesOptions from './TreegraphSeriesOptions.js';
+import type { StatesOptionsKey } from '../../Core/Series/StatesOptions';
+import type SVGAttributes from '../../Core/Renderer/SVG/SVGAttributes';
+
+import NodesComposition from '../NodesComposition.js';
+import OrganizationSeries from '../Organization/OrganizationSeries.js';
+import SankeyColumnComposition from '../Sankey/SankeyColumnComposition.js';
+import Series from '../../Core/Series/Series.js';
+import SeriesRegistry from '../../Core/Series/SeriesRegistry.js';
+import SVGRenderer from '../../Core/Renderer/SVG/SVGRenderer.js';
+import TreegraphNode from './TreegraphNode.js';
+import TU from '../TreeUtilities.js';
+const { prototype: { symbols } } = SVGRenderer;
+const { getLevelOptions } = TU;
+const {
+    seriesTypes: {
+        sankey: { prototype: sankeyProto }
+    }
+} = SeriesRegistry;
+
+import H from '../../Core/Globals.js';
+import U from '../../Core/Utilities.js';
+const { merge, pick, addEvent, stableSort } = U;
+
+import './TreegraphLayout.js';
```

Comment:
Shouldn't we instead import the exported part from this file (and create this exported part if does not exist)?

Reply:
same as in https://github.com/highcharts/highcharts/blob/master/ts/Series/Networkgraph/Networkgraph.ts#L58

I wanted to be consistent. What do you think @pawelfus?

Reply:
In Networkgraph/PackedBubble, we need some higher-level API, to support multiple series subscribing (adding its nodes and links) to one running layout algorithm. Can we create a chart with multiple tree series (roots)? If yes, then are these rendered separately (each runs its own positioning algorithm, one-by-one) or use a shared process?

Reply:
Now, each series creates its own `layoutAlgorithm` instance of the class, and each series is rendered separately. This also means, that the series will overlap, if there is more than one `treegraph` series.